instrumentation/netty/netty-4-common/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/netty/common/client/NettyClientInstrumenterFactory.java

```diff
@@ -25,16 +26,24 @@ public NettyClientInstrumenterFactory(String instrumentationName) {
   public Instrumenter<HttpRequestAndChannel, HttpResponse> createHttpInstrumenter() {
     NettyHttpClientAttributesExtractor httpClientAttributesExtractor =
         new NettyHttpClientAttributesExtractor();
+    NettyNetClientAttributesExtractor netClientAttributesExtractor =
+        new NettyNetClientAttributesExtractor();
 
     return Instrumenter.<HttpRequestAndChannel, HttpResponse>newBuilder(
             GlobalOpenTelemetry.get(),
             instrumentationName,
             HttpSpanNameExtractor.create(httpClientAttributesExtractor))
         .setSpanStatusExtractor(HttpSpanStatusExtractor.create(httpClientAttributesExtractor))
         .addAttributesExtractor(httpClientAttributesExtractor)
+        // in case of netty client instrumentation we're using 2 net attributes extractors:
+        // 1. the common one will extract net attributes on start of the operation; in case of
+        // timeouts or other connection issues netty may return null when calling
+        // Channel.remoteAddress() at the end of processing
+        // 2. the client one will extract full net attributes at the end of processing - it should
+        // be the fully resolved address at this point in time
         .addAttributesExtractor(new NettyCommonNetAttributesExtractor())
-        // TODO: add peer extractor attributes once Net*AttributesExtractors are refactored
-        // .addAttributesExtractor(PeerServiceAttributesExtractor.create(netClientAttributesExtractor))
+        .addAttributesExtractor(netClientAttributesExtractor)
+        .addAttributesExtractor(PeerServiceAttributesExtractor.create(netClientAttributesExtractor))
```

Comment:
`net.peer.service` won't be captured on read timeout? what about alt idea I think touched on yesterday of adding `remoteAddress` into `HttpRequestAndChannel`, and then using that in `NettyNetClientAttributesExtractor`?

Reply:
Hmm, I'll try that out - I haven't thought about `peer.service` in the timeout case.